marrow/mailer/address.py

```diff
@@ -143,7 +143,7 @@ def __init__(self, addresses=None, encoding="utf-8"):
 	
 	def __repr__(self):
 		if not self: return "AddressList()"
-		return f"AddressList(\"{", ".join([str(i) for i in self])}\")")
+		return f"AddressList(\"{', '.join(str(i) for i in self)}\")"
```

Comment:
A ha, no, the original comprehension was actually correct! 😁 Good catch on the missing parenthesis, though; that's my bad. What you've replaced it with is a list comprehension, which, while still functional, stops and then creates a concrete list from the result of a generator comprehension adding the memory overhead of a true list to each evaluation. The original was the bare [generator expression](https://docs.python.org/3.9/glossary.html#index-20). Join accepts ["iterable objects"](https://docs.python.org/3/library/stdtypes.html#str.join), which both lists and generators qualify as.

Reply:
Yup, nope, my dyslexia got the best of me here. 🙇 Apologies. Sometimes, to go faster, I need to slow down.